app/src/androidTest/java/com/saucelabs/mydemoapp/android/view/activities/WebviewTest.java

```diff
@@ -0,0 +1,74 @@
+package com.saucelabs.mydemoapp.android.view.activities;
+
+import androidx.lifecycle.Lifecycle;
+import androidx.test.espresso.contrib.RecyclerViewActions;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.saucelabs.mydemoapp.android.R;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+
+@RunWith(AndroidJUnit4.class)
+public class WebviewTest {
+
+    String url;
+
+    @Rule
+    public ActivityScenarioRule<MainActivity> activityRule
+            = new ActivityScenarioRule<>(MainActivity.class);
+
+    @Before
+    public void setUp(){
+        activityRule.getScenario().moveToState(Lifecycle.State.RESUMED);
+        url="https://www.google.com";
+
+    }
+
+    @Test
+    public void withoutUrlTest() {
+        // Open Drawer to click on navigation.
+        onView(withId(R.id.menuIV)).perform(click());
```

Comment:
With Espresso you should be able to jump to a specific screen. All testcases below have the same flow of first opening the menu and then go to the login. Can this be altered?